components/BottomDrawer/src/MDCBottomDrawerTransitionController.h

```diff
@@ -0,0 +1,40 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@class MDCBottomAppBarView;
+
+// TODO: (larche) Cite spec article.
+/**
+ A controller for scrubbable scrolling (aka dragging) vertical pan gesture-based mnodal transitions
+ from the bottom of the screen initiated in an MDCBottomAppBarView.
+ */
+@interface MDCBottomDrawerTransitionController: UIPercentDrivenInteractiveTransition
+    <UIViewControllerTransitioningDelegate>
+
+/**
+ Initializes the controller and adds a pan gesture recognizer to the bottom app bar.
+
+ @param bottomAppBar The bottom app bar that initiates and reacts to the presentation of the
+ drawer.
+ */
+- (nonnull instancetype)initWithBottomAppBar:(nonnull MDCBottomAppBarView *)bottomAppBar
```

Comment:
It looks like this will always require an MDC app bar. Is it possible that some apps will wan this to launch off a custom view? Is there anything strictly requiring this?

Reply:
Who will own this view?  I would expect it to be a subview of the presenting controller (MDCBottomAppBarController?).  Will we require the presented view controller to be of a certain type?

Reply:
`MDCBottomAppBarController` will indeed own the view and the drawer. So agreed with @ajsecord and @jgunaratne in a working session.